test/serialization/DefaultSerializersLiveTest.js

```diff
@@ -15,102 +15,138 @@
         });
-    });
 
-    it('utf8 characters test with surrogates', function () {
-        return map.put('key', '\u0040\u0041\u01DF\u06A0\u12E0\uD834\uDF06').then(function () {
-            return RC.executeOnController(cluster.id, _generateGet('key'), 1);
-        }).then(function (response) {
-            return expect(response.result.toString()).to.equal('\u0040\u0041\u01DF\u06A0\u12E0\u{1D306}');
+        it('number', function () {
+            return map.put('a', 23).then(function () {
+                return RC.executeOnController(cluster.id, _generateGet('a'), 1);
+            }).then(function (response) {
+                return expect(Number.parseInt(response.result.toString())).to.equal(23);
+            })
         });
-    });
 
-    it('utf8 sample string test', function () {
-        return map.put('key', 'Iñtërnâtiônàlizætiøn').then(function () {
-            return RC.executeOnController(cluster.id, _generateGet('key'), 1);
-        }).then(function (response) {
-            return expect(response.result.toString()).to.equal('Iñtërnâtiônàlizætiøn');
+        it('array', function () {
+            return map.put('a', ['a', 'v', 'vg']).then(function () {
+                return RC.executeOnController(cluster.id, _generateGet('a'), 1);
+            }).then(function (response) {
+                return expect(response.result.toString()).to.equal(['a', 'v', 'vg'].toString());
+            })
         });
-    });
 
-    it('number', function () {
-        return map.put('a', 23).then(function () {
-            return RC.executeOnController(cluster.id, _generateGet('a'), 1);
-        }).then(function (response) {
-            return expect(Number.parseInt(response.result.toString())).to.equal(23);
-        })
-    });
+        it('emoji string test on client', function () {
+            return map.put('key', '1⚐中💦2😭‍🙆😔5').then(function () {
+                return map.get('key');
+            }).then(function (response) {
+                return expect(response).to.equal('1⚐中💦2😭‍🙆😔5');
+            });
+        });
+
+        it('utf8 characters test on client', function () {
+            return map.put('key', '\u0040\u0041\u01DF\u06A0\u12E0\u{1D306}').then(function () {
+                return map.get('key');
+            }).then(function (response) {
+                return expect(response).to.equal('\u0040\u0041\u01DF\u06A0\u12E0\u{1D306}');
+            });
+        });
+
+        it('utf8 characters test on client with surrogates', function () {
+            return map.put('key', '\u0040\u0041\u01DF\u06A0\u12E0\uD834\uDF06').then(function () {
+                return map.get('key');
+            }).then(function (response) {
+                return expect(response).to.equal('\u0040\u0041\u01DF\u06A0\u12E0\u{1D306}');
+            });
+        });
+
+        // TODO: remove the check in future when string serialization in client protocol changes
+        if (stringSerializationPolicy === StringSerializationPolicy.LEGACY) {
```

Comment:
Instead of conditionally defining the test cases, please define the test and use `this.skip()` inside the test. That way, we can see there are skipped tests inside the file from the reports. Otherwise, these are not visible at all unless somebody scans test files.

Reply:
Thanks for noticing it! Makes sense. I'll change this test.